app/controllers/api/v1/callbacks_controller.rb

```diff
@@ -1,11 +1,15 @@
 class Api::V1::CallbacksController < Api::BaseController
   skip_before_action :authenticate_with_token
-  before_action :verify_firebase_token, :validate_country_code, :validate_mobile_number, :validate_uid, :validate_id_token
-  before_action :validate_name, only: :mobile_sign_up
+  before_action :validate_country_code, :validate_mobile_number, :validate_uid, :validate_id_token, :verify_firebase_token
 
   def mobile_login
-    @user = User.find_by!(mobile: mobile_provider_params[:mobile])
-    render_success
+    @user = User.find_by(mobile: mobile_provider_params[:mobile])
+
+    if @user.present?
+      render_success
+    else
+      render json: {}, status: :unauthorized
```

Comment:
@nronas I just realised this is a contract change and requires frontend to change code. Let me check the feasibility since we are so close to launch.